drafts/bitcartel-nu0-tx-format/draft.rst

```diff
@@ -0,0 +1,207 @@
+
+    0x80000003 & 0x7FFFFFFFF = 0x00000003 = 3
+
+Existing code can continue to check the transaction version using greater than comparison operators::
+
+    if (tx.nVersion >= 3) {
+      for (int js = 0; js < joinsplits; js++) {
+
+Existing tests can continue to set tx.nVersion to zero as an error condition::
+
+    mtx.nVersion = 0;
+    // https://github.com/zcash/zcash/blob/59de56eeca6f9f6f7dc1841630d53676075242a5/src/gtest/test_mempool.cpp#L99
+
+    EXPECT_CALL(state, DoS(100, false, REJECT_INVALID, "bad-txns-version-too-low", false)).Times(1);
+    // https://github.com/zcash/zcash/blob/30d3d2dfd438a20167ddbe5ed2027d465cbec2f0/src/gtest/test_checktransaction.cpp#L99
+
+Currently, the nVersion field is a public member variable which can be accessed directly.  As part of implementing Overwinter, the nVersion field will be made private with access restricted to using getters, e.g.::
+
+    bool isLegacyFormat()        // return true if the most significant bit of nVersion is unset
+    int32 getRawVersion()        // return version field, -2147483645
+    int32 getVersion()           // return version number, -2147483645 (Legacy), 3 (Overwinter)
+
+
+Forwards Compatibility
+----------------------
+
+A branch may support many transaction version formats.  For example:
+
+* Zcash reference implementation, branch "Zcash", versions 3, 4.
+* Fork of Zcash, branch "Clone", versions 3, 4*
+
+Where transaction format version 4* for the "Clone" branch might be substantially different from the expected transaction format version 4 for the "Zcash" branch.
+
+Given forwards compatibility, we want the "Zcash" branch nodes to accept transaction version 4, whilst rejecting version 4* transactions which are intended only for the "Clone" branch.
+
+To achieve this, Overwinter requires a transaction to include a branch ID, to explicitly state which branch of the network this transaction is intended for.
+
+Overwinter introduces a new signature hashing scheme which includes the branch ID, but by including the branch ID into the transaction format, clients can quickly reject transactions during deserialization without having to check signatures.
+
+A simple way to filter transactions might look like this::
+
+    if (tx.branchID != CLIENT_BRANCH_ID) { ... }
+    
+However given that a branch may support a set of transaction version formats, we should implement such that we can write code like::
+
+    if (isBranchSupported(tx.getBranchID())) { ... }
+
+    if (tx.isSupportedBranch()) { ... }
+    
+Overwinter will introduce a method for developers to easily specify and update a map of supported branch IDs and transaction versions which can be easily accessed throughout the system.
+
+Deployment
+==========
+
+This proposal will be deployed with the Overwinter network upgrade.
+
+Testnet:
+
+Mainnet:
```

Comment:
We can already say that the deployment will be at a particular block height for each of testnet and mainnet, even if we don't know what those block heights are.